src/spec/test/typing/TypeCheckingTest.groovy

```diff
@@ -44,6 +44,24 @@ class TypeCheckingTest extends StaticTypeCheckingTestCase {
         '''
     }
 
+    void testTypeCheckedEachWithIndex(){
+        //TODO: tags?
+        assertScript '''
+            @groovy.transform.TypeChecked
```

Comment:
No need to add this annotation within a `StaticTypeCheckingTestCase`.  The test case takes care of this for you and the subclass also applies static compilation.